---
title: "FISH 546 Final Writeup"
author: "Aidan Coyle"
date: "3/4/2021"
bibliography: "references.bib"
csl: "ecology.csl"
output:
  html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r citations, echo = FALSE}
# Set up citations
library("knitcitations")
options("citation_format" = "pandoc")
```

## Background - _Hematodinium sp._ and _Chionoecetes bairdi_. 

Before describing the details of my project, we'll examine the system! First, we'll start with the host - the Tanner crab, or _Chionoecetes bairdi_. 

_C. bairdi_ is a true crab native to the North Pacific. It is a widely harvested species, and has quite a substantial commercial, recreational, and subsistence value in Alaska. Within the state, it ranges from southeastern Alaska up to the mid-Bering Sea. Its populations are well-monitored, with annual trawl surveys and pot surveys throughout its Alaskan range. It also has a wide range of parasites, including leeches, chitin-eating bacteria, fungal infections, and the subject of this analysis - the parasitic dinoflagellate _Hematodinium sp._

_Hematodinium sp._ is a broad parasite of many crustaceans, including crabs, amphipods, lobsters, and shrimp. Widely distributed throughout waters north of the equator, it infects many valuable commercial species, including Alaskan snow and Tanner crab (_Chionoecetes spp._). First observed in western Europe in the 1930s, _Hematodinium sp._ was found in Tanner crab of southeast Alaska in 1985 `r citep("10.3354/dao003195")`. Since then, it has become endemic to the region. _Hematodinium_ progresses slowly within _Chionoecetes spp._ with several months between presumed infection and mortality, but is fatal in nearly 100% of cases `r citep("10.3354/dao064253")`. As the infection progresses, _Hematodinium_ proliferates throughout the host, resulting in tissue breakdown and respiratory damage `r citep("10.1007/s42995-020-00061-z")`. The hemolymph becomes an opaque white color, giving the crab a distinct 'cooked' appearance. The crab flesh also becomes bitter and chalky, making the crab unpalatable. This condition is known as Bitter Crab Syndrome (BCS). 

Alaskan _Chionoecetes_ stocks are already impacted by _Hematodinium sp._, both through direct mortality and the reduction of marketable live crab. However, there is concern that climate change could result in increased rates of _Hematodinium sp._ prevalence. Though there is evidence of a linkage between _Hematodinium_ infection rates and warming waters, the links to climate appear to be specific to hosts and regions. Therefore, specifically examining the response of _Chionoecetes bairdi_ and _Hematodinium sp._ to elevated temperatures will shed light on how this important crab species will be impacted by climate change, thus improving fishery management.

## Experiment and Data Source

To investigate the response of _Chionoecetes bairdi_ and _Hematodinium sp._ to temperature, Grace Crandall carried out a project in collaboration with NOAA examining gene expression in infected and uninfected crab at different temperatures. Crab were held at ambient, low, and elevated temperatures for 17 days, with hemolymph samples taken on days 0, 2, and 17. These samples were then preserved and the RNA was sequenced. In a previous analysis (performed by Grace Crandall), differences in gene expression between infected and uninfected crab was examined, along with the impact of temperature on _Chionoecetes bairdi_ gene expression. However, no analysis has yet been performed on the impact of temperature on gene expression of _Hematodinium sp._ My goal was to perform that analysis and determine how gene expression differed between _Hematodinium sp._ from hosts kept at different temperatures.

There are three nuances that complicate this goal. First, samples from Day 0 were taken when all crabs were still held at ambient temperatures - water temperature only began to change after hemolymph samples were taken. Therefore, samples from Elevated Day 0 and Lowered Day 0 groups were, in fact, taken from crabs that had been experiencing ambient temperatures. Second, during the experiment, there was a mass mortality event within the Elevated treatment group. As a result, we lack any samples from Elevated Day 17 crab.And finally, only one sample was sequenced from each day in the Lowered treatment group. Without replication, analyzing differential expression between Lowered and Elevated/Ambient conditions is impossible.

#### Individual and Pooled Libraries

Both individual (single sample) and pooled (multiple samples sequenced together) libraries were produced from this experiment. However, in this analysis, we restricted pairwise comparisons to be between individual samples only. Importantly, **individual libraries were taken from the same crab over time**. 

|Condition   | Day 0 | Day 2 | Day 17|
|------------|-------|-------|-------|
|**Elevated**|G, H, I|G, H, I|       |
|**Ambient** |A, B, C|A, B, C|A, B, C|
|**Lowered** |E      |E      |E      |

Table 1: Individual libraries. Letters correspond to IDs of the crab they originate from. For instance, all samples labeled B originate from the same crab.

# Analysis

## Analysis Background:

We will run the following R packages:

**DESeq2**: Performs differential expression analysis using negative binomial distribution model. [Documentation available here](https://bioconductor.org/packages/release/bioc/html/DESeq2.html).

**GO-MWU**: Measures the enrichment of gene ontology (GO) categories using a Mann-Whitney U (MWU) test. [Documentation available here](https://github.com/z0on/GO_MWU)

Scripts are each given a two-digit number. The first indicates the analysis they belong to, the second indicates the step in the analysis. For instance, the 5th step of the 4th analysis would be numbered 45_scriptname

Analyses are numbered as follows:

1) Analysis 1: Pairwise comparisons of expression with cbai_transcriptomev2.0
2) Analysis 2: Attempting to determine taxa by BLASTing against publicly-available sequences and genomes
3) Analysis 3: Pairwise comparisons of expression with hemat_transcriptomev1.6


## Analysis 1: Pairwise Comparisons for Unfiltered Transcriptome

This analysis utilized a previously-assembled transcriptome, referred to both in this analysis as cbai_transcriptomev2.0 (as per the original Roberts Lab notation) and as cbaihemat_transcriptomev2.0 (my notation - to clarify that it is not filtered by taxa). This transcriptome is available on the [Roberts Lab Genomic Resources page](https://robertslab.github.io/resources/Genomic-Resources/).

In this analysis, the goal was to get all differentially-expressed genes, and then determine the taxa of each - whether it belonged to _Hematodinium_ or _C. bairdi_. However, this proved quite difficult, resulting in the ongoing re-doing of much of this with a _Hematodinium_-specific transcriptome.


#### Functions: hematodinium_analysis_functions.R

This file contains all custom-built functions used in this analysis


#### 11_download_libraries_run_kallisto.ipynb

This script does the following:

1) Download individual libraries, pooled libraries, and transcriptome (links available in script)

2) Create a kallisto index from transcriptome

3) Use kallisto to pseudoalign libraries to index, thus quantifying counts for all libraries. 

4) Use a perl script from the [Trinity pipeline](https://github.com/trinityrnaseq/trinityrnaseq/wiki/Trinity-Transcript-Quantification) to group kallisto alignments into matrices. Each matrix represents samples for one pairwise comparison of individual libraries. We performed the following pairwise comparisons. Unless otherwise specified, only individual libraries were used in pairwise comparisons:

- Ambient Day 0 vs. Ambient Day 2
- Ambient Day 0 vs. Ambient Day 17
- Ambient Day 2 vs. Ambient Day 17
- Ambient Day 2 vs. Elevated Day 2
- Elevated Day 0 vs. Elevated Day 2
- Elevated Day 2 vs. Ambient Day 0 + 2
- Ambient Day 0 + 2 + 17 + Elevated Day 0 + Lowered Day 0 vs. Elevated Day 2. For this sample, both individual and pooled libraries were used. This compared all possible samples taken at ambient temperatures to all possible samples taken at elevated temperatures.

#### 12_kallisto_to_deseq_to_accessionIDs.Rmd

This script does the following:

1) Analyzes the Trinity matrices of kallisto transcript counts we just built with DESeq2, which provides statistics on the difference in expression between conditions. This outputs the following:
- Tables on statistical difference in expression for all sequences
- Tables on statistical difference in expression for differentially-expressed sequences
- MA plots of expression and log2 fold change
- MA plots shrunken using apeglm
- Dispersion estimates
- PCA plot

2) Turns the DESeq2 output tables into newline-separated files of UniProt accession IDs, which will later be modified to create input for GO-MWU

To do each of these steps, it utilizes custom-built functions available in hematodinium_analysis_functions.R

#### 13_uniprot_to_GO.Rmd

This script has a single purpose - it is used to obtain gene ontology (GO) terms from accession IDs. It utilizes a custom-built function available in hematodinium_analysis_functions.R

Using this script also requires you to have a recently-downloaded copy of the UniProt accession database (including GO terms). 

#### 14_eliminate_duplicates.ipynb

This short script utilizes a perl script available from the [GO-MWU repo](https://github.com/z0on/GO_MWU) to eliminate all repeated genes, thus finishing one of the two inputs necessary for GO-MWU - a two-column tab-separated table of genes and GO terms

#### 15_GO-MWU_prep.Rmd

Another short script that utilizes a custom-built function available in hematodinium_analysis_functions.R. This takes a DESeq2 output file of all genes and a BLASTx annotation of the transcriptome used, and produces the second input necessary to run GO-MWU - a 2-column CSV of accession IDs and unadjusted p-values with a header line. 

#### 16_running_GO-mwu

This folder contains all the files necessary to run [GO-MWU](https://github.com/z0on/GO_MWU), including the script we will actually run the analysis from - 06_running-GO-MWU.R

For each pairwise comparison, this produces the following:

- A hierarchical clustering tree of significant GO categories. Font size indicates level of enrichment, colors indicate whether up-regulated or down-regulated

- Data table containing GO terms for each gene and measures of interest

- Dissimilarity matrix of GO categories

- Results of Mann-Whitney U (MWU) test

## Analysis 2: Determining Taxa for Analysis 1

#### 21_obtaining_TPM_for_DEGs.Rmd

This script takes a list of genes that are differentially expressed between two groups and cross-references it with the kallisto indices to produce a table containing all transcript and gene IDs for all inputted DEGs, along with all TPM counts for individual kallisto quantifications for those DEGs.

The purpose of this was to obtain DEG counts for a future analysis

#### 22_DEG_blast.ipynb

This script BLASTs all DEGs from Analysis 1 for our two main comparisons (Amb. Day 2 vs. Elev. Day 2 and Elev. Day 0 vs. Elev. Day 2) against databases containing all publicly-available Alveolata (the taxa containing _Hematodinium sp_) nucleotide sequences and all Arthropoda (the taxa containing _C. bairdi_) nucleotide sequences. The purpose of these BLASTs was to determine whether a particular DEG was more likely to be _C. bairdi_ or _Hematodinium_. 

#### 23_blastn_analysis.Rmd

This script analyzes the results of our BLASTs performed in 22_DEG_blast.ipynb. It determined the number of genes with a lower e-value when BLASTed against the Alveolata database than the Arthropoda database and vice-versa. This should indicate whether each DEG is more likely to be _C. bairdi_ or _Hematodinium sp._. However, it only examines DEGs and doesn't account for sequences with relatively close e-values. 

#### 24_ncbi_genome_blasts.ipynb

This script tries to account for the flaws of our previous BLAST and BLAST analysis by widening the focus from DEGs to the full transcriptome and BLASTing against specific genomes rather than all sequences in a taxa. It takes cbai_transcriptomev2.0 (which, again, is not filtered by taxa), and BLASTs it against genomes closely-related to the two species that have been downloaded from the NCBI database. The purpose is to determine which sequences in our transcriptome originate from _C. bairdi_ and which originate from _Hematodinium sp._ by comparing the e-values. However, regardless of the e-values we set as a bar, we simply didn't get enough matches in our BLASTs - of the 1.4 million sequences in cbai_transcriptomev2.0, only 610,000 were matched to either genome, despite setting an e-value bar of 10^-2 (even double-counting sequences that matched to both). Due to this, we moved on to Analysis 2, which looks for differential expression among sequences that match to a previously-assembled _Hematodinium_ transcriptome

## Analysis 3: Pairwise comparisons of expression with hemat_transcriptomev1.6

All these scripts are essentially identical to their counterparts from Analysis 1, only differing in their input files. However, rather than building libraries against a kallisto index from cbai_transcriptomev2.0, we are building them against hemat_transcriptomev1.6, which is taxa-filtered to presumably only include _Hematodinium sp._ sequences. More extensive documentation is available [here](https://robertslab.github.io/sams-notebook/2021/03/08/Transcriptome-Assembly-Hematodinium-Transcriptomes-v1.6-and-v1.7-with-Trinity-on-Mox.html)

There is one script that is an exception to this, described below:

#### 30_hemat1.6_indexcreation.ipynb

In Analysis 1, a BLASTx index for cbai_transcriptomev2.0 had already been created, so we simply used that to match sequence IDs and accession IDs. However, no such index had already been created for hemat_transcriptomev1.6. Therefore, this script was created as a predecessor to the subsequent analysis. It BLASTs hemat_transcriptomev1.6 against the UniProt/SwissProt database to produce that index file

# Results

## Analysis 1: Pairwise Comparisons for Unfiltered Transcriptome

This section focuses largely on visualizations. For tables, see the following for [DESeq2](https://github.com/afcoyle/hemat_bairdi_transcriptome/tree/main/graphs/DESeq2_output/cbaihemat_transcriptomev2.0), or for [GO-MWU](https://github.com/afcoyle/hemat_bairdi_transcriptome/tree/main/output/GO-MWU_output/cbaihemat_transcriptomev2.0)

Number of differentially-expressed sequences, as determined by DESeq2:

- Ambient Day 2 vs. Elevated Day 2: 2067

- Elevated Day 0 vs. Elevated Day 2: 338

PCA plot (produced by DESeq2):

- [Ambient Day 2 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/cbaihemat_transcriptomev2.0/amb2_vs_elev2_indiv/PCA_plot.png)

- [Elevated Day 0 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/cbaihemat_transcriptomev2.0/elev0_vs_elev2_indiv/PCA_plot.png)

MA plot (produced by DESeq2):

- [Ambient Day 2 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/cbaihemat_transcriptomev2.0/amb2_vs_elev2_indiv/normalizedcts_v_log2foldchange.png)

- [Elevated Day 0 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/cbaihemat_transcriptomev2.0/elev0_vs_elev2_indiv/normalizedcts_v_log2foldchange.png)

GO-MWU Enrichment Plot:

- [Ambient Day 2 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/GOMWU_output/cbaihemat_transcriptomev2.0/amb2_vs_elev2_indiv.png)

- [Elevated Day 0 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/GOMWU_output/cbaihemat_transcriptomev2.0/elev0_vs_elev2_indiv.png)

Our GO-MWU results show that despite a lower number of DEGs, there was much more pathway enrichment in Elevated Day 0 vs. Elevated Day 2 than in Ambient Day 2 vs. Elevated Day 2. This is likely due to the decreased variability in gene expression provided by examining the same crabs and parasites over time, rather than comparing different crab/parasite systems, and may indicate that in scenarios where adding additional pairwise comparisons become onerous, it may be optimal to focus on Elevated Day 0 vs. Elevated Day 2. 

Our results are also not filtered by taxa. Given that the goal is to examine differential expression within _Hematodinium sp._ specifically, this means that we need a way to differentiate _Hematodinium sp._ sequences from other sequences. 

## Analysis 2: Exploratory Analysis and Determining Taxa

We first BLASTed DEGs against a database of all Alveolata and Arthropoda sequences in an attempt to determine whether they were more closely related to _C. bairdi_ or _Hematodinium sp._ However, we ran into a few issues. First, and most notably, it only examines DEGs. As a result, we broadened our focus to BLAST the full transcriptome (cbai_transcriptomev2.0), and rather than BLASTing against databases of all sequences within the taxa, we BLASTed against the most closely-related genome to both _Hematodinium sp._ and _C. bairdi_. For _C. bairdi_, we found an extremely closely-related genome - _C. opilio_, which regularly hybridizes with _C. bairdi_. However, we had no such luck with _Hematodinium sp._, and chose the genome of _Amoebophrya sp._. 

After this second round of BLASTs, we [found an unacceptably low number of sequences were being matched, even after dropping our e-values substantially](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/scripts/24_ncbi_genome_blasts.ipynb). As a result, we moved on to Analysis 3, which solves the problem of determining taxa by simply only examining _Hematodinium sp._ sequences

## Analysis 3: Pairwise Comparisons for Hematodinium-Only Transcriptome

This section focuses largely on visualizations. For tables, see the following for [DESeq2](https://github.com/afcoyle/hemat_bairdi_transcriptome/tree/main/graphs/DESeq2_output/hemat_transcriptomev1.6), or for [GO-MWU](https://github.com/afcoyle/hemat_bairdi_transcriptome/tree/main/output/GO-MWU_output/hemat_transcriptomev1.6)

Number of differentially-expressed sequences, as determined by DESeq2:

- Ambient Day 2 vs. Elevated Day 2: 1

- Elevated Day 0 vs. Elevated Day 2: 4

PCA plot (produced by DESeq2):

- [Ambient Day 2 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/hemat_transcriptomev1.6/amb2_vs_elev2_indiv/PCA_plot.png)

- Elevated Day 0 vs. Elevated Day 2: No PCA plot produced due to low overall variance

MA plot (produced by DESeq2):

- [Ambient Day 2 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/hemat_transcriptomev1.6/amb2_vs_elev2_indiv/normalizedcts_v_log2foldchange.png)

- [Elevated Day 0 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/DESeq2_output/hemat_transcriptomev1.6/elev0_vs_elev2_indiv/normalizedcts_v_log2foldchange.png)

GO-MWU Enrichment Plot:

- [Ambient Day 2 vs. Elevated Day 2](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/graphs/GOMWU_output/hemat_transcriptomev1.6/amb2_vs_elev2_indiv.png)

- Elevated Day 0 vs. Elevated Day 2: No graph possible due to low pathway enrichment

### DEG Analysis

Due to the low numbers of DEGs in Analysis 3, we can examine each one individually as part of our analysis of _Hematodinium sp._ gene expression.

#### Amb. Day 2 vs. Elev. Day 2

[**O99262**](https://www.uniprot.org/uniprot/O99252): This gene matches to the mitochondrial gene **cytochrome c oxidase subunit I, AKA COI**. This is...really interesting! I've got a bit of a history with COI - my paper on the European green crab found a linkage between thermal tolerance and COI genotype. This is obviously a bit different, as we're examining expression rather than genotype, and it's a lot more intuitive that mitochondrial activity could be linked to temperature, but still quite neat 

As a caveat, I'm not completely sure this is real. Our p-value is 5.5x10^-7, which is nice, but our p-adj is just 0.0023. And looking at other ambient-temperature libraries, 5 out of 14 had a expression level equal or lower to that of the most-expressed elevated-temperature library. But certainly an interesting result

#### Elev. Day 0 vs. Elev. Day 2

[**Q06056**](https://www.uniprot.org/uniprot/Q06056): Another mitochondrial gene! This one matches to ATP synthase F(0) complex subunit C2. P-values look a bit better for this one (pval = 9.13x10^-8, padj = 4.0 x 10^-5). Interesting that we're seeing multiple mitochondrial genes here - definitely worth a deeper dive later on.

[**O23717**](https://www.uniprot.org/uniprot/O23717). The first of two proteosome subunits- this one is proteosome subunit beta type-5-a. Looks like it cleaves peptides, and some quick Google Scholar work found that in humans, it's involved in immunosuppression and may be oncogenic. Again, p-values are solid (pval = 6.1x10^-8, padj = 4.3x10^-5).

[**P85200**](https://www.uniprot.org/uniprot/P85200). The second of two proteosome subunits - this one is proteosome regulatory subunit 6b homolog. It's involved in the ATP-dependent degradation of ubiquinated proteins (whatever that means). Again, good pvals (pval = 7.0x10^-8, padj=4.3x10^-5)



## References

``` {r references, include = FALSE}
# write bibliography
write.bibtex(file = "references.bib")

# Fix non-standard characters, otherwise they'll be unicode
# Read in new Bibtex file
fixbib <- readLines("references.bib")
# Remove brackets
fixbib <- gsub(pattern = "<[^>]|>", replace = "", x = fixbib)
# Replace 
fixbib <- gsub(pattern = "+0092", replace = "'", x = fixbib, fixed = TRUE)
writeLines(fixbib, con = "references.bib")
```